pkg/sql/logictest/testdata/logic_test/txn

```diff
@@ -631,6 +631,15 @@ BEGIN READ WRITE; COMMIT
 statement error read only not supported
 BEGIN READ ONLY
 
+# TODO(andrei): Something might be fishy here. The statement above leaves the
+# transaction in an error state, and so we need this rollback. But all the
+# errors below are parse errors, and so they don't mess with the transaction
+# state. Perhaps it'd be better if the statement above also left the session in
+# the NoTxn state. I couldn't do an analogous test with Postgres since I can't
+# trigger a BEGIN error that's not a parse error.
```

Comment:
All right, but then either take the opportunity of this PR to address this, or add a TODO and/or file a follow-up issue.